src/Constants.cpp

```diff
@@ -28,6 +28,7 @@ const uint64_t FDB_KEY_LENGTH_LIMIT = (uint64_t)1e4;
 const uint64_t FDB_VALUE_LENGTH_LIMIT = (uint64_t)1e5;
 
 const uint64_t METADATA_CACHE_SIZE = (uint64_t)100;
+const uint64_t METADATA_INVALID_VERSION = (uint64_t)-1;
```

Comment:
This should be 0 since the metadata version starts from 1. if its current value is 0, something went wrong.

Reply:
We use 0 to represent nonexistent collection. -1 to newly created collection. Errors are never represented by metadata version. I left all this behavior as it was. This patch just adds version to collection context object, semantics of version number are same as before, unless there is a bug in the patch.

Reply:
The metadataVersion is never explicitly set to 0 anywhere in the code though, the atomic operation will implicitly create it and initialize it to 0 and then add 1 to it.

Reply:
`getMetadataVersion()` returns 0 in case collection doesn’t exist.

Reply:
Right, I totally forgot that.